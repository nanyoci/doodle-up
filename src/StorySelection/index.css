@import url('https://fonts.googleapis.com/css2?family=Caveat+Brush&display=swap');

.story-boxes {
	display: grid;
	margin: auto;
	width: fit-content;
	grid-template-columns: repeat(3, 1fr);
	gap: 30px;
}

.story-box {
	display: block;
	padding: 30px;
	box-sizing: border-box;
	background: #ffffff;
	box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
	transition: all 0.3s ease-in-out;
	max-width: 250px;
}

.story-box:hover, .story-box:focus {
	background: #fff6b0;
}

.story-box img {
	width: 100%;
	height: 100%;
}

@media (max-width: 768px) {
	.story-boxes {
		grid-template-columns: repeat(2, 1fr);
	}
}

@media (max-width: 576px) {
	.story-boxes {
		grid-template-columns: repeat(2, 1fr);
		gap: 20px;
	}

	.story-box {
		padding: 20px;
	}
}
