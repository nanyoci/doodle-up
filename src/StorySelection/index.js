import React from 'react';
import { Link } from "react-router-dom";

import rabbit from '../assets/rabbit.png';
import dinosaur from '../assets/dinosaur.png';
import car from '../assets/car.png';
import './index.css';
import Page from '../Page';

export default function StorySelection() {
	return (
		<Page>
			<h1>Choose a story!</h1>
			<div className="story-boxes">
				<Link className="story-box" to="/guessing">
					<img src={rabbit} alt="Rabbit story" />
				</Link>
				<Link className="story-box" to="/drawing">
					<img src={dinosaur} alt="Dinosaur story" />
				</Link>
				<Link className="story-box" to="/story-page">
					<img src={car} alt="Car story" />
				</Link>
			</div>
		</Page>
	)
}
