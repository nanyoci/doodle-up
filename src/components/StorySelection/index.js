import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import rabbit from '../../assets/rabbit.png';
import dinosaur from '../../assets/dinosaur.png';
import car from '../../assets/car.png';
import Page from '../Page';
import { selectStories, listStories, selectStoriesLoading } from '../../redux/ducks/stories';

function StorySelection({ stories, listStories, isLoading }) {
	useEffect(() => {
		listStories()
	}, []);

	return (

		<Page isLoading={isLoading}>
			<h1>My Story Books</h1>
			<div className="story-boxes">
				{
					stories.length !== 0
						? stories.map(story =>
							<div>
								<a className="story-box" href="/">
									<img src={`${story.cover_image}`} alt="Dinosaur story" />
								</a>
								<p>{story.story_title}</p>
							</div>
						)
						: <p>No Stories completed.</p>
				}
			</div>
		</Page>
	)
}

StorySelection.propTypes = {
	stories: PropTypes.array.isRequired,
	isLoading: PropTypes.bool
};
const mapStateToProps = state => ({
	stories: selectStories(state),
	isLoading: selectStoriesLoading(state)
});

const dispatchers = {
	listStories,
};

export default connect(mapStateToProps, dispatchers)(StorySelection);

