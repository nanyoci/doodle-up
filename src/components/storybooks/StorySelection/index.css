@import url('https://fonts.googleapis.com/css2?family=Caveat+Brush&display=swap');

.story-boxes {
	display: grid;
	margin: auto;
	width: fit-content;
	grid-template-columns: repeat(3, 1fr);
	gap: 30px;
}

.story-box {
	width: 100%;
	max-width: 300px;
}

.story-box-link {
	display: block;
	box-sizing: border-box;
	background: #ffffff;
	box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
	transition: all 0.3s ease-in-out;
	width: 100%;
}

.story-box-link:hover, .story-box-link:focus {
	background: #fff6b0;
}

.story-box-link img {
	width: 100%;
	height: 100%;
}

@media (max-width: 768px) {
	.story-boxes {
		grid-template-columns: repeat(2, 1fr);
	}
}

@media (max-width: 576px) {
	.story-boxes {
		grid-template-columns: repeat(2, 1fr);
		gap: 20px;
	}
}
