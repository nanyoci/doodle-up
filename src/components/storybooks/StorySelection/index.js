import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import Page from '../../common/Page';
import { selectStories, listStories, selectStoriesLoading } from '../../../redux/ducks/stories';
import './index.css';

function StorySelection({ stories, listStories, isLoading }) {
	useEffect(() => {
		listStories()
	}, [listStories]);

	return (

		<Page isLoading={isLoading}>
			<h1>Story Books</h1>
			{
				!isLoading && (
					stories.length !== 0
						?
						<div className="story-boxes">{
							stories.map(story =>
								<div className="story-box" key={story.id}>
									<Link to={`/stories/${story.id}`} className="story-box-link">
										<img src={`${story.cover_image}`} alt={story.story_title} />
									</Link>
									<h2>{story.story_title}</h2>
								</div>
							)}
						</div>
						: <p>No stories available.</p>
				)
			}
		</Page>
	)
}

StorySelection.propTypes = {
	stories: PropTypes.array.isRequired,
	isLoading: PropTypes.bool
};
const mapStateToProps = state => ({
	stories: selectStories(state),
	isLoading: selectStoriesLoading(state)
});

const dispatchers = {
	listStories,
};

export default connect(mapStateToProps, dispatchers)(StorySelection);

