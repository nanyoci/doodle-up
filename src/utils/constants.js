export const API_URL = 'https://doodle-up-backend.herokuapp.com';