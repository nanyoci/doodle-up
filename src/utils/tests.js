import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { render } from '@testing-library/react';

import rootReducer from '../redux/rootReducer';

export const renderWithRedux = (component, initialState = {}) => {
    const store = createStore(
        rootReducer,
        initialState,
        applyMiddleware(thunk),
    )

    const renderedComponent = <Provider store={store}>{component}</Provider>;

    return {
        ...render(renderedComponent),
        store
    };
}

export const renderWithRouter = (component, route = '/') => {
    const history = createMemoryHistory({ initialEntries: [route] });
    const renderedComponent = <Router history={history}>{component}</Router>;

    return {
        ...render(renderedComponent),
        history
    };
}

export const renderWithReduxRouter = (component, initialState = {}, route = '/') => {
    const store = createStore(
        rootReducer,
        initialState,
        applyMiddleware(thunk),
    )
    const history = createMemoryHistory({ initialEntries: [route] });

    const renderedComponent = (
        <Provider store={store}>
            <Router history={history}>{component}</Router>
        </Provider>
    );

    return {
        ...render(renderedComponent),
        history,
        store
    };
}