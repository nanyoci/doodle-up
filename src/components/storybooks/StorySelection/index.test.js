import React from 'react'
import { cleanup, act } from '@testing-library/react'
import axiosMock from 'axios';

import cover from './sample/cover.png';
import StorySelection from './index.js';
import { renderWithReduxRouter } from './../../../utils/tests';

jest.mock('axios');

beforeEach(() => {
  window.HTMLMediaElement.prototype.load = () => { /* do nothing */ };
  window.HTMLMediaElement.prototype.play = () => { /* do nothing */ };
  window.HTMLMediaElement.prototype.pause = () => { /* do nothing */ };
  window.HTMLMediaElement.prototype.addTextTrack = () => { /* do nothing */ };

  axiosMock.get.mockResolvedValue({
    data: {
      results: [
        {
          cover_image: cover,
          id: "001",
          story_title: "Goldilocks and the Three Bears"
        }
      ]
    }

  });
});

afterEach(() => {
  cleanup();
  jest.clearAllMocks();
});

it('should take a snapshot', async () => {
  let asFragment = {};
  await act(async () => { ({ asFragment } = renderWithReduxRouter(<StorySelection />)) });
  expect(asFragment()).toMatchSnapshot();
})
