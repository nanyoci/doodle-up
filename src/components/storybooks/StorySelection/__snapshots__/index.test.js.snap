// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="main-content"
  >
    <audio />
    <div
      class="wrapper"
    >
      <a
        href="/"
      >
        <img
          alt="DoodleUp"
          class="logo-corner"
          src="doodle-up.svg"
        />
      </a>
      <h1>
        Story Books
      </h1>
      <div
        class="story-boxes"
      >
        <div
          class="story-box"
        >
          <a
            class="story-box-link"
            href="/stories/001"
          >
            <img
              alt="Goldilocks and the Three Bears"
              src="cover.png"
            />
          </a>
          <h2>
            Goldilocks and the Three Bears
          </h2>
        </div>
      </div>
    </div>
    <div
      class="control-buttons"
    >
      <button
        class="ui yellow massive circular icon button"
        id="signOutBtn"
      >
        <i
          aria-hidden="true"
          class="sign-out icon"
        />
      </button>
      <button
        class="ui yellow massive circular icon button"
        id="musicControlBtn"
      >
        <i
          aria-hidden="true"
          class="mute icon"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
